### ISSUE TITLE
[nsmd] Strong bug with kde file pickers

### ISSUE THREAD

**Issue by Houston4444** [original-male/non#206]
non-session-manager from kxstudio-repo
kde5, tested on kubuntu (16.04) and Neon

There is a strong bug between kde file pickers and nsm.
Under kde, when a nsm client call kde file picker, this client interface becomes totally buggy.
There is something concerning the files under kde, because dolphin (kde file manager) does the same when it's launched from nsm.
I don't know any kde program that integers the nsm protocol, but all qt5 programs call kde5 file picker on kde, so It will be very disapointting for Qtractor, Carla, Hydrogen...if we can't import audio or midi files.

I checked the env , it doesn't seems to change anything serious, there is maybe a trouble with the way nsmd launch system commands.

**Comment by original-male** [comment]
Are you sure this only happens when these programs are being run under NSM? What is nsmd's console output at the time?

**Comment by Houston4444** [comment]
I never saw this happen when running programs not in nsm.

so running non-session-manager, create session kdetest2, add kwrite and click on "open file"):
```
[non-session-manager] Starting daemon...
X_ChangeProperty: BadValue (integer parameter out of range for operation) 0x0
[nsmd] Session root is: /home/houston/NSM Sessions
NSM_URL=osc.udp://mathieubis-MS-7592:17618/
[nsmd] Listing sessions
[non-session-manager] Sending new for: testkde2
[nsmd] Creating new session "testkde2"
[non-session-manager] /nsm/server/new says Created.
[non-session-manager] /nsm/server/new says Session created
[non-session-manager] Sending add for: kwrite
[nsmd] Process has pid: 15734
[nsmd] Launching kwrite
QProcess: Destroyed while process ("testparm") is still running.
We got some errors while running testparm "Load smb config files from /etc/samba/smb.conf\nLoaded services file OK.\nWARNING: The 'netbios name' is too long (max. 15 chars).\n\n"
QProcess: Destroyed while process ("net") is still running.
QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: Permission non accordée
[nsmd] Client kwrite died unexpectedly.
[non-session-manager] Sending close.
[nsmd] Commanding attached clients to save.
[nsmd] Waiting for clients to reply to commands
[nsmd] Done waiting
[nsmd] Commanding attached clients to quit.
[nsmd] Commanding kwrite to quit
[nsmd] Waiting for killed clients to die.
[nsmd] Loop 0
[nsmd] All clients have died.
[nsmd] Done
[non-session-manager] Got message /nsm/gui/client/status from unknown client
[non-session-manager] /nsm/server/close says Closed.
[non-session-manager] Telling server to quit

```
I just saw that the window finally works after maybe...one minute. Exactly when console says that: 
QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: Permission non accordée

To compare, run kwrite and click on "open file":
```
We got some errors while running testparm "Load smb config files from /etc/samba/smb.conf\nLoaded services file OK.\nWARNING: The 'netbios name' is too long (max. 15 chars).\n\n"
QInotifyFileSystemWatcherEngine::addPaths: inotify_add_watch failed: Permission non accordée
```
And it works directly...

Does nsm send any signal to process after launch ?

**Comment by original-male** [comment]
The only signal NSM will send is SIGTERM when it wants to end the process. It's hard to see how running under NSM could be having any influence here. At least the KDE error messages may give you some clue about where the problem lies (in your SMB or network configuration?). You may be better served by taking the issue up with the KDE people.

**Comment by original-male** [comment]
One think to check (and this is important for liblo too) is, does your hostname actually resolve to an IP address?

**Comment by Houston4444** [comment]
Sorry, I don't really understand the question...you can see my hostname in NSM_URL .nsmd is launch by non-session-manager on the same machine.

**Comment by original-male** [comment]
I mean does that hostname resolve to an IP address. That is, if you open a shell and type 

host `hostname`

Do you get an address or an error?

Does kwrite behave the same way when just launched from an xterm window?

**Comment by Houston4444** [comment]
host hostname gives`Host hostname not found: 3(NXDOMAIN)`.
from xterm kwrite works normally.

**Comment by original-male** [comment]
Did you forget to type the backticks around hostname? Also, FYI, I cannot reproduce the issue on the latest Debian unstable.

**Comment by Houston4444** [comment]
on kde ?

**Comment by original-male** [comment]
Well, no, not on KDE :-)

**Comment by jrigg** [comment]
@Houston4444: I think the point original-male is making is that the problem may be in your network configuration or in KDE/QT5 networking libraries, not a bug in NSM. NSM requires a valid network configuration to work properly.

It looks from the error messages like there could be a problem with your smb configuration. Are you actually using smb (ie. to connect to a Windows system or other Samba server) or did it just get installed by default? If it's the latter you might want to uninstall it or at least disable it in the appropriate configuration file. (Caveat: I'm on Debian and don't use KDE, so treat my suggestions accordingly).

**Comment by Houston4444** [comment]
@jrigg , you're right, I uninstall samba and it works well. Now, I don't know if bug comes from kde or from samba. Have you tried it with samba installed opening a gtk dialog ?

**Comment by jrigg** [comment]
@Houston4444: My DAW installations are very minimalist and only have what is necessary to compile and use pro audio software - no desktop environment, no unnecessary network stuff etc.

It's possible the problem with Samba was a Kubuntu configuration issue rather than a software bug. I've tried Ubuntu in the past but couldn't work with it due to it doing too many things without asking me :-)
